dali/pipeline/operators/fused/new_crop_mirror_normalize.h

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_FUSED_NEW_CROP_MIRROR_NORMALIZE_H_
+#define DALI_PIPELINE_OPERATORS_FUSED_NEW_CROP_MIRROR_NORMALIZE_H_
+
+#include <cstring>
+#include <utility>
+#include <vector>
+
+#include "dali/core/common.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/pipeline/operators/operator.h"
+#include "dali/pipeline/operators/crop/crop_attr.h"
+#include "dali/kernels/scratch.h"
+
+namespace dali {
+
+template <typename Backend>
+class NewCropMirrorNormalize : public Operator<Backend>, protected CropAttr {
+ public:
+  explicit inline NewCropMirrorNormalize(const OpSpec &spec)
+      : Operator<Backend>(spec),
+        CropAttr(spec),
+        output_type_(spec.GetArgument<DALIDataType>("output_dtype")),
+        output_layout_(spec.GetArgument<DALITensorLayout>("output_layout")),
+        pad_output_(spec.GetArgument<bool>("pad_output")),
+        image_type_(spec.GetArgument<DALIImageType>("image_type")),
+        C_(IsColor(image_type_) ? 3 : 1),
```

Comment:
I'd very much like to get rid of this thing...

Reply:
done